superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx

```diff
@@ -122,8 +128,8 @@ const DatasetModal: FunctionComponent<DatasetModalProps> = ({
       if (onDatasetAdd) {
         onDatasetAdd({ id: response.id, ...response });
       }
-      window.location.href = `/chart/add?dataset=${currentTableName}`;
-      hide();
+      history.push(`/chart/add?dataset=${currentTableName}`);
+      cleanup();
```

Comment:
The previous version had a call to `onHide`. Should we keep it?

```
const hide = () => {
    setItem(LocalStorageKeys.db, null);
    clearModal();
    onHide();
};
```

Reply:
No, `onHide` only executed `history.replace`, which was supposed to remove the `#create` from URL. We only want it when we close the modal, otherwise it causes a race condition with `history.push` from line 131 (explained in https://github.com/apache/superset/pull/21683/files#r985828109)